src/resources/js/ajax-button-script.js

```diff
@@ -0,0 +1,129 @@
+/**
+ * Script for starting and stopping real-time logging of AJAX calls.
+ *
+ * @since 1.0.0
+ */
+
+jQuery( document ).ready( function ( $ ) {
+	// Variable to keep track of whether logging is active or not
+	var isLoggingActive = false;
+
+	// Container for displaying messages on the page
+	var messageContainer = $( '#ajax-message-container' );
+
+	/**
+	 * Toggles whether to listen or not for AJAX calls.
+	 *
+	 * @since 1.0.0
+	 */
+	function toggleAjaxLogging() {
+		if ( isLoggingActive ) {
+			stopAjaxLogging();
+		} else {
+			startAjaxLogging();
+		}
+	}
+
+	/**
+	 * Starts logging AJAX calls.
+	 *
+	 * @since 1.0.0
+	 */
+	function startAjaxLogging() {
+		// If logging is not active, start it
+		$( document ).on( 'ajaxSend.ajaxLogger', function ( event, jqxhr, settings ) {
+			// Append the message to the container
+			// messageContainer.append('<p>AJAX request started: ' + JSON.stringify(settings) + '</p>');
+		} );
+
+		$( document ).on( 'ajaxComplete.ajaxLogger', function ( event, jqxhr, settings ) {
+			// What is logged to the frontend
+			// messageContainer.append('<div>AJAX request completed: ' + JSON.stringify(settings) + '</div>');
+			// messageContainer.append('<div >Response: ' + jqxhr.responseText + '</div>');
+			// messageContainer.append('<div>Status: ' + jqxhr.status + '</div>');
+		} );
+
+		// Append the message to the container
+		messageContainer.append( '<p>Real-time AJAX logging started</p>' );
```

Comment:
If you are commenting out pieces, remove and use the Git history later to retrieve, helps you avoid a ton of code that doesn't need to be present.